src/pages/TrendingPage/TrendingPage.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useState } from 'react';
+
+import FeedPostList from '../../components/FeedPostsList';
+import SkeletonFeedList from '../../components/SkeletonFeedList';
+import { renderErrorAlert } from '../../components/ErrorAlert/ErrorAlert';
+import { TrendingFeedList, ErrorObject } from '../../types/trendingFeedTypes';
+import { fetchData } from '../../api';
+import { Endpoint } from '../../api/constants';
+
+const TrendingPage = () => {
+  const [posts, setPosts] = useState<TrendingFeedList | ErrorObject>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    fetchData(
+      Endpoint.TrendingFeed,
+      controller,
+      setIsLoading,
+      setPosts,
+      setIsError
+    );
+    return () => controller.abort();
+  }, []);
```

Comment:
=> I like your logic of this component, it's clear and understandable. But I can propose some adjustments.

more than 3 arguments decrease the readability of function. You can merge all 3 useStates, to one custom hook, 
and shift fetch function inside of this custom hook. customHook will return some object like {isLoading, isError, posts}
, and then, you will use only these values. So later you can use this hook for similar server fetches. 
Here is the example,what i propose: 

export const useServersRequest = (url) => {
  const [isLoading, setIsLoading] = useState(false);
  const [posts, setPosts] = useState(null);
  const [isError, setIsError] = useState(false); // or in your case use not boolean, but new Error object

  useEffect(() => {
    setIsLoading(true);

    ... fetch data
      try {
	posts =... extracting json data	
        }

        setPosts(posts);
      } catch (error) {
        setIsError(true);
      } finally {
        setIsLoading(false);
      }
    };

    fetchData();
  }, [url]);

  return { isLoading, posts, isError };

In your case, you handling exactly the error message, so you can return the required error, not a boolean value.


But if the logic for fetch is only in one component, your decision is good enough.

Reply:
Good point, thank you! I will try to implement this custom hook that you are writing about